app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java

```diff
@@ -42,6 +43,53 @@ public static DiningHallModel fromEatery(Context context, boolean hardcoded,
         return model;
     }
 
+    /**
+     * getMealIntervals returns a HashMap of all possible MealTypes and the corresponding
+     * time intervals.
+     * */
+    private HashMap<MealType, Interval> getMealIntervals() {
+        HashMap<MealType, Interval> mealIntervalMap = new HashMap<MealType, Interval>();
+        for (MealModel mealModel: getCurrentDayMenu().getAllMeals()) {
+            mealIntervalMap.put(mealModel.getType(), mealModel.getInterval());
+        }
+        return mealIntervalMap;
+    }
+
+    /**
+     * isBeforeMealType returns true if the current time comes before mealType. Otherwise, returns false.
+     * */
+    public boolean isBeforeMealType(MealType mealType, Set<MealType> set, HashMap<MealType, Interval> map) {
+        return set.contains(mealType) && LocalDateTime.now().compareTo(map.get(mealType).getEnd()) < 0;
+    }
+
+    /**
+     * getCurrentMealTypeTabIndex returns the menu tab index to be displayed, based on the
+     * current time. Note that this does NOT represent the index of the corresponding MealType enum.
+     * */
+    public int getCurrentMealTypeTabIndex() {
+        int mealTypeTabIndex = 0; // Tab index to return.
+        HashMap<MealType, Interval> mealIntervalMap = getMealIntervals();
+        Set<MealType> mealIntervalKeys = mealIntervalMap.keySet();
+        // Dinner - always last option.
+        if (isBeforeMealType(MealType.DINNER, mealIntervalKeys, mealIntervalMap)) {
+            mealTypeTabIndex = mealIntervalKeys.size() - 1;
+        }
+        // Either last or second-to-last option, depending if dinner exists.
+        if (isBeforeMealType(MealType.LUNCH, mealIntervalKeys, mealIntervalMap)) {
+            int offset = mealIntervalKeys.contains(MealType.DINNER) ? -2 : -1;
+            mealTypeTabIndex = mealIntervalKeys.size() + offset;
+        }
+        // Either first or second option, depending if breakfast exists.
+        if (isBeforeMealType(MealType.BRUNCH, mealIntervalKeys, mealIntervalMap)) {
+            mealTypeTabIndex = mealIntervalKeys.contains(MealType.BREAKFAST) ? 1 : 0;
```

Comment:
I would switch around the ordering (start with breakfast and then do the check for brunch, lunch, dinner) and use 'else ifs' instead, because currently it is just going to unnecessarily execute each if block (even though it works)